src/Sylius/Bundle/ApiBundle/CommandHandler/AddProductReviewHandler.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\CommandHandler;
+
+use ApiPlatform\Core\Api\IriConverterInterface;
+use Sylius\Bundle\ApiBundle\Command\AddProductReview;
+use Sylius\Bundle\ApiBundle\Provider\CustomerProviderInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\Review\Model\ReviewInterface;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+
+/** @experimental */
+final class AddProductReviewHandler implements MessageHandlerInterface
+{
+    /** @var CustomerProviderInterface */
+    private $customerProvider;
+
+    /** @var RepositoryInterface */
+    private $productReviewRepository;
+
+    /** @var FactoryInterface */
+    private $productReviewFactory;
+
+    /** @var IriConverterInterface */
+    private $iriConverter;
+
+    public function __construct(
+        CustomerProviderInterface $customerProvider,
+        RepositoryInterface $productReviewRepository,
+        FactoryInterface $productReviewFactory,
+        IriConverterInterface $iriConverter
+    ) {
+        $this->customerProvider = $customerProvider;
+        $this->productReviewRepository = $productReviewRepository;
+        $this->productReviewFactory = $productReviewFactory;
+        $this->iriConverter = $iriConverter;
+    }
+
+    public function __invoke(AddProductReview $addProductReview): ReviewInterface
+    {
+        /** @var ProductInterface $product */
+        $product = $this->iriConverter->getItemFromIri($addProductReview->product);
+
+        /** @var CustomerInterface|null $customer */
+        $customer = $this->customerProvider->get();
```

Comment:
I'm not convinced to use this provider in this way here, IMO the command should have information about the email of the current user (if the user is logged in), so it should behave similarly to setting a channel code on commands (in data transformer).